lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -84,6 +84,11 @@ def with_arity
         def pp_names
           @names.length == 1 ? "##{@names.first}" : description_of(@names)
         end
+
+        def get_method_on_object(object, method_name)
+          # Can't use object.method since it might be overridden
+          Kernel.instance_method(:method).bind(object).call(method_name)
```

Comment:
We have a method in rspec-support that handles this plus several other edge cases:

https://github.com/rspec/rspec-support/blob/92e571c80b23d6ba643ec2d8372126e36deddf88/lib/rspec/support.rb#L35-L74

Please use `RSpec::Support.method_handle_for` instead.